### DISCUSSION TITLE
iOS 在并行队列中异步调用 pagView.flush()，会有问题没，有 UI 操作吗

### DISCUSSION THREAD

**Discussion by AppHock** [Tencent/libpag#2092] [Q&A] [answered]
背景：tableView有多个 cell，每个 cell 上只有 pagView，屏幕上会存在多个 cell，然后 cell 滚动在屏幕中心的才开始播放，否则就暂停。在 cellForRow 中设置 pagView 的 pagFile，设置 pagView 暂停，并调用 flush

问题：当前屏幕中心有 pagView 正在播放时，滚动时加载 新的cell 时触发 cellForRow 加载pagView，并调用 flush，会产生卡顿

尝试解决：我将 cellForRow 中，只将pagView.flush()代码 放在自己创建的并行队列中异步调用，会有问题没？flush 我是在非主线车中调用，该方法里面会存在 iOS 中 UI 操作api调用吗

```
self.pagFile = PAGFile.load(pagPath)
self.pagView.setComposition(self.pagFile)
// pagView 可能不支持离屏渲染，如果是隐藏状态调用 flush，不会上效果到 pagView 上
self.pagView.isHidden = false
// 异步刷新 pag，避免造成滑动卡顿的同时，解决因 cell 复用导致旧的 pag 和当前 pag 画面切换问题
pagFlushQueue.async {
    self.pagView.flush()
}
```

**Comment by kevingpqi123** [comment, marked as answer]
可以这样处理， PAGView 播放的过程中就是在异步线程中调用的 flush，当需要上屏的时候，内部会切换至主线程

如果不涉及占位图替换或文本编辑，可以试下 PAGImageView 组件, PAGImageView 支持渲染缓存至本地，只有第一次加载的时候会卡一些，缓存完成后就是加载本地缓存
